packages/firebase_ui_storage/doc/getting-started.md

```diff
@@ -0,0 +1,124 @@
+# Getting started with Firebase UI Storage
+
+Firebase UI Storage is a set of Flutter widgets and utilities designed to help you build and integrate your user interface with Firebase Storage.
+
+## Installation
+
+Install dependencies
+
+```sh
+flutter pub add firebase_core firebase_storage firebase_ui_storage
+```
+
+Download Firebase project config
+
+```sh
+flutterfire configure
+```
+
+## Configuration
+
+This section will walk you through the configuration process of the Firebase UI Storage
+
+### macOS
+
+If you're building for macOS, you will need to add an entitlement for either read-only access if you only upload files:
+
+```xml
+  <key>com.apple.security.files.user-selected.read-only</key>
+  <true/>
+```
+
+or read/write access if you want to be able to download files as well:
+
+```xml
+  <key>com.apple.security.files.user-selected.read-write</key>
+  <true/>
+```
+
+Make sure to add network client entitlement as well:
+
+```xml
+<key>com.apple.security.network.client</key>
+<true/>
+```
+
+### Initialize Firebase
+
+```dart
+import 'package:firebase_core/firebase_core.dart';
+import 'firebase_options.dart';
+
+Future<void> main() async {
+    WidgetsFlutterBinding.ensureInitialized();
+
+    await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
+}
+```
+
+### Configure Firebase UI Storage
+
+`FirebaseUIStorage` has a top-level configuration, that reduces amount of widget properties needed
+to be passed to the widget constructors.
+
+```dart
+import 'package:firebase_storage/firebase_storage.dart';
+
+Future<void> main() async {
+  // configuration steps from above
+
+  final storage = FirebaseStorage.instance;
+
+  final config = FirebaseUIStorageConfiguration(
+    storage: storage,
+    uploadRoot: storage.ref('flutter-tests'),
+    namingPolicy: UuidFileUploadNamingPolicy(), // optional, will generate a UUID for each uploaded file
+  );
+
+  await FirebaseUIStorage.configure(config);
+}
+```
+
+Available file naming policies:
+
+- [`KeepOriginalNameUploadPolicy`] – keeps the original file name
+- [`KeepPathUploadPolicy`] - keeps the original file path
+- [`UuidFileUploadNamingPolicy`] – generates a UUID for each uploaded file
+
+[`KeepOriginalNameUploadPolicy`]: https://pub.dev/documentation/firebase_ui_storage/latest/firebase_ui_storage/KeepOriginalNameUploadPolicy-class.html
+[`KeepPathUploadPolicy`]: https://pub.dev/documentation/firebase_ui_storage/latest/firebase_ui_storage/KeepPathUploadPolicy-class.html
+[`UuidFileUploadNamingPolicy`]: https://pub.dev/documentation/firebase_ui_storage/latest/firebase_ui_storage/UuidFileUploadNamingPolicy-class.html
+
+If you ever need to override this top-level configuration, you can do so by using `FirebaseUIStorageConfigOverride`:
```

Comment:
Could we possibly mention the default here?

Reply:
aa8ee60bf6fecc14d7b0606527f01c138433074b